### ISSUE TITLE
Calling deprecated methods – warn?

### ISSUE THREAD

**Issue by dvdplm** [paritytech/jsonrpsee#564]
Consider an RPC server where some methods are marked as deprecated:

```rust
	#[rpc(client, server, namespace = "foo")]
	pub trait Rpc {
		#[method(name = "foo")]
		async fn async_method(&self, param_a: u8, param_b: String) -> RpcResult<u16>;

		#[deprecated(since="0.5.0", note="please use `new_method` instead")]
		#[method(name = "bar")]
		fn sync_method(&self) -> RpcResult<u16>;
       }
```

Ideally, using the generated Rust client should inform/warn the user when `client.sync_method()` is called. Preferably at compile time.

e.g. https://github.com/paritytech/substrate/pull/8783#discussion_r749319311

**Comment by niklasad1** [comment]
That's good idea I like it

**Comment by lexnv** [comment]
PR for deprecated methods: https://github.com/paritytech/jsonrpsee/pull/570

**Comment by niklasad1** [comment]
Closed by #570
